### PR TITLE
[NET-11043] crd: support request normalization and header match options to prevent L7 intentions bypass

### PR DESCRIPTION
**Pull request by zalimeni**
### Changes proposed in this PR ###  

Supports changes in https://github.com/hashicorp/consul/pull/21816 with CRD updates:

- [crd: support L7 intentions header match contains and ignoreCase](https://github.com/hashicorp/consul-k8s/commit/2ce1e0b740d0306948555444aa2b24a7b993a2f2)
- [crd: support mesh http.incoming.requestNormalization](https://github.com/hashicorp/consul-k8s/commit/10a45b5a77cb103031b28cd3a6c3deca344c34c0)
- [crd: generate updated CRDs](https://github.com/hashicorp/consul-k8s/commit/d8d289e60a40f0273beb9e3c73bff3b0cb80b55a)
- [crd: remove requirement for mesh http.sanitizeXForwardedClientCert](https://github.com/hashicorp/consul-k8s/commit/a48fe67cd28fd1fecbd85e09f0e9b3381f54315f)

### How I've tested this PR ###
* Added unit tests
* Manually spun up cluster in `kind` and validated both changes are reflected in xDS / config when created as CRDs

### How I expect reviewers to test this PR ###
👀 + spot check me on the CRD update process in case I missed a step. It's been a while since I've done one of these.

### Checklist ###
- [x] Tests added
- [x] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: charts/consul/templates/crd-meshes.yaml [hunk 1]
```diff
@@ -66,10 +66,56 @@ spec:
               http:
                 description: HTTP defines the HTTP configuration for the service mesh.
                 properties:
+                  incoming:
```
Comment (zalimeni) [inline charts/consul/templates/crd-meshes.yaml:69]: Docs here and below copied from `consul` changes. Subject to change based on feedback from Docs team.

File: control-plane/config/crd/bases/consul.hashicorp.com_meshes.yaml [hunk 2]
```diff
@@ -62,10 +62,56 @@ spec:
...
+                              `/` or `\` characters or upstream services are configured to differentiate between escaped and unescaped slashes.
+                              The default value of this option is empty, which is equivalent to `IMPLEMENTATION_SPECIFIC_DEFAULT`. See Envoy
+                              docs for more information on available options.
+                            type: string
+                        type: object
+                    type: object
                   sanitizeXForwardedClientCert:
                     type: boolean
-                required:
-                - sanitizeXForwardedClientCert
```
Comment (zalimeni) [inline control-plane/config/crd/bases/consul.hashicorp.com_meshes.yaml:68]: Changed by adding `omitempty` to this field; now that we have multiple fields, requiring this one no longer makes sense. I don't know of any validation we require that a config entry has _some_ explicit fields (`service-defaults` is allowed to be empty), so I think just removing this is fine.

File: control-plane/go.mod [hunk 3]
```diff
@@ -2,6 +2,11 @@ module github.com/hashicorp/consul-k8s/control-plane
 
 replace github.com/hashicorp/consul-k8s/version => ../version
 
+// temporary for local dev
+replace (
+	github.com/hashicorp/consul/api => ../../consul-enterprise/api
+)
```
Comment (zalimeni) [inline control-plane/go.mod:8]: To be removed once `api` is cut and updated in this file

File: control-plane/catalog/metrics/metrics.go [hunk 4]
```diff
@@ -1,3 +1,6 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```
Comment (zalimeni) [inline control-plane/catalog/metrics/metrics.go:2]: Seems this and `metrics_util.go` were missed on the last codegen update

**Review by jmurret — APPROVED** [review]
LGTM.  Nice work!

**Comment by zalimeni** [comment]
Rebased and squashed for merge

**Comment by zalimeni** [comment]
Re-initiating backports to fix v2 tests that were disabled on `main` but not 1.4-1.6.

**Pull request merged**